### PR TITLE
#24 2 upsert feed back

### PR DESCRIPTION
**Pull request by alesyalis**
Update feedback, if is not in the database, then create a new

### CODE REVIEW

File: Recrutify/.editorconfig [hunk 1]
```diff
@@ -26,3 +26,15 @@ dotnet_diagnostic.S1118.severity = none
 
 # S101: Types should be named in PascalCase
 dotnet_diagnostic.S101.severity = none
+
+# CA1051: Do not declare visible instance fields
+dotnet_diagnostic.CA1051.severity = none
+
+# SA1401: Fields should be private
+dotnet_diagnostic.SA1401.severity = none
```
Comment (elizavetapavliv) [inline Recrutify/.editorconfig:34]: Remove

File: Recrutify/.editorconfig [hunk 2]
```diff
@@ -26,3 +26,15 @@ dotnet_diagnostic.S1118.severity = none
...
+
+# SA1202: Elements should be ordered by access
+dotnet_diagnostic.SA1202.severity = none
```
Comment (elizavetapavliv) [inline Recrutify/.editorconfig:37]: Remove

File: Recrutify/Recrutify.DataAccess/Repositories/CandidateRepository.cs [hunk 3]
```diff
@@ -11,5 +16,38 @@ public CandidateRepository(IOptions<MongoSettings> options)
...
+                   new BsonDocument("feedback.Type", feedback.Type),
+                })),
+            };
+            var updateOptions = new UpdateOptions { ArrayFilters = arrayFilters };
+
+            var updateResult = await GetCollection().UpdateOneAsync(filter, updateDefinition, updateOptions);
+            if (updateResult.ModifiedCount == 0)
+            {
+                arrayFilters = new List<ArrayFilterDefinition>
+            {
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.DataAccess/Repositories/CandidateRepository.cs:44]: Format it correctly, brackets should be on level deeper

File: Recrutify/Recrutify.Host/Controllers/CandidateController.cs [hunk 4]
```diff
@@ -30,5 +31,18 @@ public async Task<ActionResult<CandidateDTO>> CreateAsync(CandidateCreateDTO can
             var result = await _candidateService.CreateAsync(candidateCreateDTO);
             return Created(string.Empty, result);
         }
+
+        [HttpPut]
+        public async Task<ActionResult> UpsertAsync(Guid id, Guid projectId, FeedbackDTO feedbackDto)
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Host/Controllers/CandidateController.cs:36]: UpsertFeedbackAsync

File: Recrutify/Recrutify.DataAccess/Repositories/CandidateRepository.cs [hunk 5]
```diff
@@ -11,5 +16,38 @@ public CandidateRepository(IOptions<MongoSettings> options)
             : base(options)
         {
         }
+
+        public async Task UpsertAsync(Guid id, Guid projectId, Feedback feedback)
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.DataAccess/Repositories/CandidateRepository.cs:20]: UpsertFeedbackAsync

**Comment by elizavetapavliv** [comment]
Merge main and resolve conflicts

File: Recrutify/.editorconfig [hunk 6]
```diff
@@ -26,3 +26,15 @@ dotnet_diagnostic.S1118.severity = none
 
 # S101: Types should be named in PascalCase
 dotnet_diagnostic.S101.severity = none
```
Comment (elizavetapavliv) [inline Recrutify/.editorconfig:28]: Remove

File: Recrutify/Recrutify.DataAccess/Repositories/CandidateRepository.cs (hunk 5 above)
Comment (alesyalis) [reply Recrutify/Recrutify.DataAccess/Repositories/CandidateRepository.cs]: done

File: Recrutify/Recrutify.Host/Controllers/CandidateController.cs (hunk 4 above)
Comment (alesyalis) [reply Recrutify/Recrutify.Host/Controllers/CandidateController.cs]: done

File: Recrutify/.editorconfig (hunk 2 above)
Comment (alesyalis) [reply Recrutify/.editorconfig]: done

File: Recrutify/.editorconfig (hunk 1 above)
Comment (alesyalis) [reply Recrutify/.editorconfig]: done

File: Recrutify/.editorconfig (hunk 6 above)
Comment (alesyalis) [reply Recrutify/.editorconfig:28]: done

File: Recrutify/Recrutify.DataAccess/Repositories/CandidateRepository.cs (hunk 3 above)
Comment (alesyalis) [reply Recrutify/Recrutify.DataAccess/Repositories/CandidateRepository.cs]: done

File: Recrutify/.editorconfig [hunk 7]
```diff
@@ -24,5 +24,11 @@ dotnet_diagnostic.SA0001.severity = none
 # S1118: Utility classes should not have public constructors
 dotnet_diagnostic.S1118.severity = none
 
+# CA1051: Do not declare visible instance fields
+dotnet_diagnostic.CA1051.severity = none
+
+# S1854: Unused assignments should be removed
+dotnet_diagnostic.S1854.severity = none
```
Comment (elizavetapavliv) [inline Recrutify/.editorconfig:31]: remove

File: Recrutify/Recrutify.Host/Configuration/Profiles/ProjectProfile.cs [hunk 8]
```diff
@@ -13,7 +13,6 @@ public ProjectProfile()
                 .ForMember(dest => dest.Id, conf => conf.MapFrom(src => Guid.NewGuid()));
             CreateMap<Project, ProjectDTO>().ReverseMap();
             CreateMap<StaffDTO, Staff>().ReverseMap();
-            CreateMap<PrimarySkill, PrimarySkillDTO>();
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Host/Configuration/Profiles/ProjectProfile.cs:16]: how your primary-skills endpoint will work?

Comment (alesyalis) [reply Recrutify/Recrutify.Host/Configuration/Profiles/ProjectProfile.cs:16]: done

File: Recrutify/.editorconfig (hunk 7 above)
Comment (alesyalis) [reply Recrutify/.editorconfig]: done

**Review by elizavetapavliv — APPROVED** [review]

**Pull request merged**